tests/performance/event_generator/src/event_generator/main.py

```diff
@@ -0,0 +1,181 @@
+                        help='Path to YAML config file')
+    args = parser.parse_args()
+
+    validate_parameters(args)
+
+    return args
+
+
+def validate_parameters(args: argparse.Namespace) -> None:
+    """Validate the command line arguments provided.
+
+    Args:
+        args (argparse.Namespace): The arguments provided to the command line.
+
+    Raises:
+        ValueError: If any required arguments are missing or invalid.
+
+    Ensures that the 'config' argument is provided and that it points to a valid file.
+    """
+    if not args.config:
+        raise ValueError("Configuration file must be provided.")
+    if not os.path.isfile(args.config):
+        raise ValueError("Invalid configuration file.")
+
+
+def get_logcollector_generator(file_config: dict[str, Any]) -> LogEventGenerator:
+    """Create and return a LogEventGenerator instance configured according to the provided file configuration.
+
+    Args:
+        file_config (dict): Configuration settings specific to the logcollector module. It must include
+                            necessary parameters like module, path, operations, rate, max_file size, and
+                            template path.
+
+    Returns:
+        LogEventGenerator: An instance of LogEventGenerator configured as per the file_config.
+
+    Raises:
+        ValueError: If any required parameters are missing in file_config for logcollector.
+    """
+    required_params = ['module', 'path', 'operations',
+                       'rate', 'max_file_size', 'template_path']
+    if not all(param in file_config for param in required_params):
+        raise ValueError(
+            "Missing required config parameters for logcollector.")
+    return LogEventGenerator(
+        rate=file_config['rate'],
+        path=file_config['path'],
+        operations=file_config['operations'],
+        max_file_size=file_config['max_file_size'],
+        template_path=file_config['template_path']
+    )
+
+
+def get_syscheck_generator(file_config: dict[str, Any]) -> SyscheckEventGenerator:
+    """Create and return a SyscheckEventGenerator instance based on the provided file configuration.
+
+    Args:
+        file_config (dict): Configuration settings specific to the syscheck module. It should include
+                            necessary parameters like module, path, operations, and rate.
+
+    Returns:
+        SyscheckEventGenerator: An instance of SyscheckEventGenerator configured according to the file_config.
+
+    Raises:
+        ValueError: If any required parameters are missing in file_config for syscheck.
+    """
+    required_params = ['module', 'path', 'operations', 'rate']
+    if not all(param in file_config for param in required_params):
+        raise ValueError("Missing required config parameters for syscheck.")
+    return SyscheckEventGenerator(
+        rate=file_config['rate'],
+        path=file_config['path'],
+        operations=file_config['operations']
+    )
```

Comment:
Code repetition, try to merge them into one event factory function.

Reply:
Fixed here: https://github.com/wazuh/wazuh-qa/pull/5697/commits/f294135c2147822d452dda551ed1c762268598c9